AppDelegate.mm

```diff
@@ -160,43 +164,54 @@ - (void)applicationDidFinishLaunching:(NSNotification*)notification {
               if (task.terminationStatus != 0) {
                 [fileManager removeItemAtPath:tempDir error:nil];
 
-                dispatch_async(dispatch_get_main_queue(), ^{
-                  showErrorModal(
-                      [NSString stringWithFormat:@"Failed to extract: %i", task.terminationStatus]);
-                });
+                // If extraction fails, it's likely to be a permission issue let's rerun the program
+                // as root
+
+                if (geteuid() != 0) {
+                  STPrivilegedTask* task = [STPrivilegedTask new];
+                  [task setLaunchPath:[NSBundle.mainBundle executablePath]];
```

Comment:
Doing this at this point would require re-downloading, right? What if we check if the bundle is writable on launch and if not, immediately relaunch as admin?

Reply:
Makes sense, I wanted to avoid explicit permission checks but re-download isn't a great experience.

Reply:
Updated this to try creating a test folder